hypervisor/lib/crypto/mbedtls/sha256.c

```diff
@@ -203,47 +204,53 @@ int32_t mbedtls_sha256_update_ret( mbedtls_sha256_context *ctx,
                                const uint8_t *input,
                                size_t ilen )
 {
-    int32_t ret;
+    int32_t ret = 0;
     size_t fill;
     uint32_t left;
-
-    if( ilen == 0 )
-        return( 0 );
-
-    left = ctx->total[0] & 0x3F;
-    fill = 64 - left;
-
-    ctx->total[0] += (uint32_t) ilen;
-    ctx->total[0] &= 0xFFFFFFFF;
-
-    if( ctx->total[0] < (uint32_t) ilen )
-        ctx->total[1]++;
-
-    if( left && ilen >= fill )
-    {
-        memcpy_s( (void *) (ctx->buffer + left), fill, input, fill );
-
-        if( ( ret = mbedtls_internal_sha256_process( ctx, ctx->buffer ) ) != 0 )
-            return( ret );
-
-        input += fill;
-        ilen  -= fill;
-        left = 0;
-    }
-
-    while( ilen >= 64 )
-    {
-        if( ( ret = mbedtls_internal_sha256_process( ctx, input ) ) != 0 )
-            return( ret );
-
-        input += 64;
-        ilen  -= 64;
+    const uint8_t *data = input;
+    size_t len = ilen;
+
+    if ( (len != 0U) && (data != NULL)) {
+        left = ctx->total[0] & 0x3FU;
+        fill = 64U - left;
+
+        ctx->total[0] += (uint32_t)len;
+        ctx->total[0] &= 0xFFFFFFFFU;
+
+        if ( ctx->total[0] < (uint32_t)len ) {
+            ctx->total[1]++;
+        }
+
+        if ( (left != 0U) && (len >= fill) ) {
+            memcpy_s( (void *)&ctx->buffer[left], fill, data, fill );
+
+            ret = mbedtls_internal_sha256_process( ctx, ctx->buffer );
+            if (ret == 0) {
+                data += fill;
+                len  -= fill;
```

Comment:
remove the extra space, same for the others

Reply:
done